presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSpatialPartitioningInternalAggregation.java

```diff
@@ -93,6 +96,27 @@ public void test(int partitionCount)
         assertEquals(groupValue, expectedValue);
     }
 
+    @Test
+    public void testEmptyPartitionException()
+    {
+        InternalAggregationFunction function = getFunction();
+
+        Block geometryBlock = GEOMETRY.createBlockBuilder(null, 0).build();
+        Block partitionCountBlock = BlockAssertions.createRLEBlock(10, 0);
+        Page page = new Page(geometryBlock, partitionCountBlock);
+
+        AccumulatorFactory accumulatorFactory = function.bind(Ints.asList(0, 1, 2), Optional.empty());
+        Accumulator accumulator = accumulatorFactory.createAccumulator();
+        accumulator.addInput(page);
+        try {
+            getFinalBlock(accumulator);
+            fail("Should fail creating spatial partition with no rows.");
+        }
+        catch (PrestoException e) {
+            assertEquals(e.getErrorCode(), INVALID_FUNCTION_ARGUMENT.toErrorCode());
```

Comment:
nit: it would be nice to confirm the error message as well

Reply:
It seems like the actual wording of the message is an implementation detail?  I'm happy to check the string, but it seems like something we'd just copy, and have to maintain separately if changed the error message.  I'm happy to do it, tho!

Reply:
This is a user-visible message, hence, I like to make sure it makes. I think copying is ok here. I don't feel strongly about this though.